src/Shared/Cosmos/CosmosUtils.cs

```diff
@@ -49,12 +49,12 @@ internal static CosmosConnectionInfo ParseConnectionString(string connectionStri
 
         // Strip out the database and container from the connection string in order
         // to tell if we are left with just AccountEndpoint.
-        if (connectionBuilder.TryGetValue("Database", out var _))
+        if (connectionBuilder.TryGetValue("Database", out var databaseValue))
         {
             connectionBuilder["Database"] = null;
         }
 
-        if (connectionBuilder.TryGetValue("Container", out var _))
+        if (connectionBuilder.TryGetValue("Container", out var containerValue))
```

Comment:
Why does this need to affect the `DbContext` configuration?

Reply:
It doesn't affect the `DbContext` at all at the moment. This is just shared code that ends up getting used in the non-EF scenario.

The EF-specific overloads currently disregard the container value if it's set. This will work correctly assuming we can implement the [IModelInitializedConvention](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.conventions.imodelinitializedconvention?view=efcore-9.0) API you referenced.

Reply:
What do you mean by "correctly"? I don't understand why this needs to have any impact on the `DbContext`. I don't think the additional code we discussed should be added

Reply:
> What do you mean by "correctly"? I

Correctly here means that the container name associated with the connection string we are parsing is set as the container for entities registered in the context.

> I don't think the additional code we discussed should be added

Is your recommendation for the EF integration that we disregard the container name and always assume the default container name is resolved from the name of the DbContext.

Reply:
> Is your recommendation for the EF integration that we disregard the container name and always assume the default container name is resolved from the name of the DbContext.

Yes, because the default container name is used only if the container name for a given entity type is not set. So changing the default might not have any effect if all container names are configured explicitly. Also, it wouldn't have any effect if the user is relying on a pre-compiled model as by the time this code runs the model is already built.